src/data/polynomial/basic.lean

```diff
@@ -793,9 +775,9 @@ variables [ring R]
 instance : has_int_cast R[X] := ⟨λ n, of_finsupp n⟩
 
 instance : ring R[X] :=
-function.injective.ring to_finsupp to_finsupp_injective
-  to_finsupp_zero to_finsupp_one to_finsupp_add to_finsupp_mul to_finsupp_neg to_finsupp_sub
-  (λ _ _, to_finsupp_smul _ _) (λ _ _, to_finsupp_smul _ _) to_finsupp_pow (λ _, rfl) (λ _, rfl)
+to_finsupp_injective.ring to_finsupp to_finsupp_zero to_finsupp_one to_finsupp_add to_finsupp_mul
+  to_finsupp_neg to_finsupp_sub (λ _ _, to_finsupp_smul _ _) (λ _ _, to_finsupp_smul _ _)
+  to_finsupp_pow (λ _, rfl) (λ _, rfl)
```

Comment:
I don't see real reason to prefer the new version over the old, so let's keep the diff clean:
```suggestion
function.injective.ring to_finsupp to_finsupp_injective
  to_finsupp_zero to_finsupp_one to_finsupp_add to_finsupp_mul to_finsupp_neg to_finsupp_sub
  (λ _ _, to_finsupp_smul _ _) (λ _ _, to_finsupp_smul _ _) to_finsupp_pow (λ _, rfl) (λ _, rfl)
```